### PR TITLE
Fix many bindable transfer cases using leased bindables

### LINKED ISSUE #4008: Exiting multiplayer room keeps the room's ruleset active
The ruleset should be reset to the user's last choice when exiting rooms.

### LINKED ISSUE #4037: Exiting a room/multiplayer leaves game in "no beatmaps available" state
- Enter room where you don't have the beatmap
- Exit room
- Exit multiplayer

Observe: main menu does not play music and music controller shows "no beatmaps available".

### PR DESCRIPTION
**Pull request by peppy**
- Closes #4037.
- Closes #4008.
- Can be used to handle #4018.
- [x] Depends on ppy/osu-framework#2102.

Removes ugly `ForcefullySet` calls.

### CODE REVIEW

File: osu.Game/osu.Game.csproj [hunk 1]
```diff
@@ -18,7 +18,7 @@
     <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="2.2.1" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.Core" Version="2.2.1" />
     <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
-    <PackageReference Include="ppy.osu.Framework" Version="2019.117.0" />
+    <PackageReference Include="ppy.osu.Framework" Version="0.0.8033" />
```
Comment (VPeruS) [inline osu.Game/osu.Game.csproj]: Version from local build?

Comment (peppy) [reply osu.Game/osu.Game.csproj]: from appveyor build (see osu-framework wiki page for instructions)

Comment (VPeruS) [reply osu.Game/osu.Game.csproj]: Then build scipt missing adding appveyor source(in case of CI build)

**Comment by peppy** [comment]
Closed via #4191.

**Pull request closed without merging**